### PR TITLE
Add a tool that will do re.sub on a per column basis.

### PR DESCRIPTION
**Pull request by blankenberg**

### CODE REVIEW

File: tools/column_regex_substitution/test-data/column_regex_substitution_out.tabular [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.	Firmicutes	Proteobacteria	Actinobacteria	Deinococcus-Thermus	Bacteroidetes	Synergistetes	Plactomycetes
+
+expected	50	30	10	5	5	0	0
+
```
Comment (nsoranzo) [inline tools/column_regex_substitution/test-data/column_regex_substitution_out.tabular]: I haven't checked the Python code, but this test output seems wrong: it has a new empty line after each original line and the substitution with a dot happened only on the first row.

Comment (blankenberg) [inline tools/column_regex_substitution/test-data/column_regex_substitution_out.tabular]: Nice catch. Fixed. Thanks.

**Comment by bgruening** [comment]
@blankenberg I like this wrapper it looks solid, but I would like to take the opportunity to guide to to this one and the entire family: https://github.com/bgruening/galaxytools/blob/master/tools/text_processing/text_processing/replace_text_in_column.xml

I'm open to split them up into several repositories and for sure they need some love, but I guess going and supporting the `coreutils` here makes a lot of sense. They scale very nicely and offer a lot of features.

**Comment by bgruening** [comment]
@blankenberg how do we want to proceed here?

**Comment by blankenberg** [comment]
@bgruening I'm a big fan of using coreutils. But theres a few features that I need that are offered by this tool that aren't currently handled by ```replace_text_in_column```. I can definitely see a path forward with enhancing replace_text_in_column to match abilities, but I wouldn't have any time to work on it for a while.

**Comment by bgruening** [comment]
@blankenberg please feel free to merge if you feel like this is the way forward. I'm just concerned about an additional tool that will be deprecated soon and just confuses users.

**Comment by jmchilton** [comment]
xref https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/regex_find_replace/regex_tabular.xml

File: tools/column_regex_substitution/column_regex_substitution.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<tool id="column_regex_substitution" name="Column substitution" version="0.0.1">
+    <description>by regular expressions</description>
+    <stdio>
+        <exit_code range="1:" />
+        <exit_code range=":-1" />
+    </stdio>
+    <version_command interpreter="python">column_regex_substitution.py --version</version_command>
+    <command interpreter="python"><![CDATA[
```
Comment (bgruening) [inline tools/column_regex_substitution/column_regex_substitution.xml]: Interpreter is deprecated.

File: tools/column_regex_substitution/column_regex_substitution.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        #import pipes
+        column_regex_substitution.py
+        --input "${input_tabular}"
```
Comment (bgruening) [inline tools/column_regex_substitution/column_regex_substitution.xml]: single-quotes

File: tools/column_regex_substitution/column_regex_substitution.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        --pattern ${pipes.quote( str( $pattern ).replace('\\','\\\\').decode( 'string_escape' ) ) or '""'}
+        --replacement ${ pipes.quote( str( $replacement ).replace('\\','\\\\').decode( 'string_escape' ) ) or '""' }
+        --columns "${ $columns or "" }"
```
Comment (bgruening) [inline tools/column_regex_substitution/column_regex_substitution.xml]: single-quotes

File: tools/column_regex_substitution/column_regex_substitution.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+<tool id="column_regex_substitution" name="Column substitution" version="0.0.1">
+    <description>by regular expressions</description>
+    <stdio>
```
Comment (bgruening) [inline tools/column_regex_substitution/column_regex_substitution.xml]: Could also be replaced by `<command detect_errors="exit_code">`

File: tools/column_regex_substitution/column_regex_substitution.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        sys.exit(0)
+
+    if None in [ options.pattern, options.replacement, options.output ]:
+        parser.print_help()
+        sys.exit(1)
+
+    pattern = options.pattern
+    replacement = options.replacement
+    column_offset = int( options.column_offset )
+    print "Pattern: %s\nReplacement: %s" % ( repr( pattern ), repr( replacement ) )
```
Comment (bgruening) [inline tools/column_regex_substitution/column_regex_substitution.py]: This is not python3 and no python2 dependency is defined.

**Comment by mvdbeek** [comment]
If there's still some interest we can reconsider this, but I'll close this for now.

**Pull request closed without merging**